service/matching/matchingEngine.go

```diff
@@ -48,6 +48,10 @@ import (
 	"github.com/uber/cadence/common/types"
 )
 
+// If sticky poller is not seem in last 10s, we treat it as sticky worker unavailable
+// This seems aggressive, but the default sticky schedule_to_start timeout is 5s, so 10s seems reasonable.
+const _stickyPollerUnavailableWindow = 10 * time.Second
```

Comment:
kinda implies it should be based on the default/configured sticky timeout.

if a user did have a very-long sticky schedule_to_start, would it be worth changing this time?  like say they set it to 5 days, to allow sequentially processing a large batch job on a single worker (say it's CPU bound).

Reply:
Is sticky schedule to start configurable?

Reply:
Does that affect polling interval? If they set it to 5 days, and the tasklist didn't receive any poller for a period, we assume the worker is down and return the error. 10s should be fine even for long timeout.
Is the polling interval something configurable?

Reply:
ah, yea, I guess "time in queue" doesn't really affect "time spent after being started" does it.  👍